Fix numerical error in test_highway

test_highway is unstable:

```
04:07:05 Traceback (most recent call last):
04:07:05   File "/work/chainer/chainer/testing/condition.py", line 60, in <lambda>
04:07:05     lambda: f(ins, *args[1:], **kwargs),
04:07:05   File "/work/chainer/tests/chainer_tests/links_tests/connection_tests/test_highway.py", line 85, in test_backward_gpu
04:07:05     self.check_backward(cuda.to_gpu(self.x), cuda.to_gpu(self.gy))
04:07:05   File "/work/chainer/tests/chainer_tests/links_tests/connection_tests/test_highway.py", line 75, in check_backward
04:07:05     eps=1e-2, atol=3.2e-3, rtol=1e-2)
04:07:05   File "/work/chainer/chainer/gradient_check.py", line 273, in check_backward
04:07:05     testing.assert_allclose(gp, p.grad, atol=atol, rtol=rtol)
04:07:05   File "/work/chainer/chainer/testing/array.py", line 24, in assert_allclose
04:07:05     x, y, atol=atol, rtol=rtol, verbose=verbose)
04:07:05   File "/usr/local/lib/python3.4/dist-packages/numpy/testing/utils.py", line 1297, in assert_allclose
04:07:05     verbose=verbose, header=header)
04:07:05   File "/usr/local/lib/python3.4/dist-packages/numpy/testing/utils.py", line 665, in assert_array_compare
04:07:05     raise AssertionError(msg)
04:07:05 AssertionError: 
04:07:05 Not equal to tolerance rtol=0.01, atol=0.0032
04:07:05 
04:07:05 (mismatch 100.0%)
04:07:05  x: array([-0.075854,  0.649984,  0.      ], dtype=float32)
04:07:05  y: array([-0.075854,  0.680829,  0.      ], dtype=float32)
```

--- a/tests/chainer_tests/links_tests/connection_tests/test_highway.py
+++ b/tests/chainer_tests/links_tests/connection_tests/test_highway.py
@@ -4,11 +4,11 @@ import numpy
 
 import chainer
 from chainer import cuda
+from chainer import functions
 from chainer import gradient_check
 from chainer import links
 from chainer import testing
 from chainer.testing import attr
-from chainer.testing import condition
 
 
 class TestHighway(unittest.TestCase):
@@ -21,7 +21,7 @@ class TestHighway(unittest.TestCase):
         self.gy = numpy.random.uniform(
             -1, 1, (5, self.in_out_size)).astype(numpy.float32)
         self.link = links.Highway(
-            self.in_out_size)
+            self.in_out_size, activate=functions.tanh)
 
         Wh = self.link.plain.W.data
         Wh[...] = numpy.random.uniform(-1, 1, Wh.shape)
@@ -39,13 +39,10 @@ class TestHighway(unittest.TestCase):
         self.Wt = Wt.copy()  # fixed on CPU
         self.bt = bt.copy()  # fixed on CPU
 
-        a = self.relu(self.x.dot(Wh.T) + bh)
+        a = numpy.tanh(self.x.dot(Wh.T) + bh)
         b = self.sigmoid(self.x.dot(Wt.T) + bt)
         self.y = (a * b +
                   self.x * (numpy.ones_like(self.x) - b))
-
-    def relu(self, x):
-        return numpy.maximum(x, 0, dtype=x.dtype)
 
     def sigmoid(self, x):
         half = x.dtype.type(0.5)
@@ -57,12 +54,10 @@ class TestHighway(unittest.TestCase):
         self.assertEqual(y.data.dtype, numpy.float32)
         testing.assert_allclose(self.y, y.data)
 
-    @condition.retry(3)
     def test_forward_cpu(self):
         self.check_forward(self.x)
 
     @attr.gpu
-    @condition.retry(3)
     def test_forward_gpu(self):
         self.link.to_gpu()
         self.check_forward(cuda.to_gpu(self.x))
@@ -74,12 +69,10 @@ class TestHighway(unittest.TestCase):
              self.link.transform.W, self.link.transform.b),
             eps=1e-2, atol=3.2e-3, rtol=1e-2)
 
-    @condition.retry(3)
     def test_backward_cpu(self):
         self.check_backward(self.x, self.gy)
 
     @attr.gpu
-    @condition.retry(3)
     def test_backward_gpu(self):
         self.link.to_gpu()
         self.check_backward(cuda.to_gpu(self.x), cuda.to_gpu(self.gy))

